src/evict/evict_lru.c

```diff
@@ -1144,25 +1162,31 @@ retry:	while (slot < max_entries && ret == 0) {
 		incr = false;
 	}
 
-	if (dhandle_locked) {
-		__wt_spin_unlock(session, &conn->dhandle_lock);
-		dhandle_locked = false;
-	}
-
 	/*
 	 * Walk the list of files a few times if we don't find enough pages.
 	 * Try two passes through all the files, give up when we have some
 	 * candidates and we aren't finding more.
 	 */
-	if (cache->pass_intr == 0 && ret == 0 &&
-	    slot < max_entries && (retries < 2 ||
+	if (slot < max_entries && (retries < 2 ||
 	    (retries < 10 &&
 	    (slot == queue->evict_entries || slot > start_slot)))) {
 		start_slot = slot;
 		++retries;
 		goto retry;
```

Comment:
I don't think there is any performance impact: this should be the rare case where the eviction server gets to the end of the handle list without finding a handle it wants to visit, then decides to restart.  I made the change so I could use `WT_ERR` rather than `break` for getting out of the loop: tracing through the `ret == 0` checks made the code brittle.